crates/sui-graphql-rpc/src/types/suins_registration.rs

```diff
@@ -350,7 +397,97 @@ impl NameService {
             .to_rust()
             .ok_or_else(|| Error::Internal("Malformed Suins Domain".to_string()))?;
 
-        Ok(Some(field.value))
+        let domain = Domain(field.value);
+
+        // We attempt to resolve the domain to a record, and if it fails, we return None.
+        // That way we can validate that the name has not expired and is still valid.
+        let Some(_) = Self::resolve_to_record(db, config, &domain).await? else {
+            return Ok(None);
+        };
+
+        Ok(Some(domain.0))
+    }
+
+    /// Queries for the NameRecord and its parent, if it is supplied.
+    /// Returns the NameServiceQuery, which contains the NameRecord (option), ParentNameRecord (option), and the
+    /// latest checkpoint's timestamp.
+    async fn resolve_name_record_query(
+        db: &Db,
+        record_id: SuiAddress,
+        parent_record_id: Option<SuiAddress>,
+    ) -> Result<Option<NameServiceQuery>, Error> {
+        let mut object_ids = vec![record_id];
+        let mut name_record = None;
+        let mut parent_name_record = None;
+
+        if let Some(id) = parent_record_id {
+            object_ids.push(id);
+        }
+
+        // prepare the filter for the query.
+        let filter = ObjectFilter {
+            object_ids: Some(object_ids),
+            ..Default::default()
+        };
+
+        let response = db
+            .execute_repeatable(move |conn| {
+                // get latest checkpoint.
+                let checkpoint = Checkpoint::latest_checkpoint(conn)?;
```

Comment:
in that sense, do we rly need `checkpoint::latest_checkpoint`? We can just do `checkpoint::query` and pass in the optional `checkpoint_viewed_at`